prow/cmd/deck/main_test.go

```diff
@@ -416,3 +419,145 @@ func TestHelp(t *testing.T) {
 }
+
+func TestListProwJobs(t *testing.T) {
+	templateJob := &prowapi.ProwJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "first",
+			Namespace: "prowjobs",
+			Labels: map[string]string{
+				"foo": "bar",
+			},
+		},
+		Spec: prowapi.ProwJobSpec{
+			Refs: &prowapi.Refs{
+				Org:  "org",
+				Repo: "repo",
+			},
+		},
+	}
+
+	var testCases = []struct {
+		name        string
+		prowJobs    []func(*prowapi.ProwJob) runtime.Object
+		listErr     bool
+		hiddenRepos sets.String
+		hiddenOnly  bool
+		expected    sets.String
+		expectedErr bool
+	}{
+		{
+			name:        "list error results in filter error",
+			listErr:     true,
+			expectedErr: true,
+		},
+		{
+			name: "no hidden repos returns all prowjobs",
+			prowJobs: []func(*prowapi.ProwJob) runtime.Object{
+				func(in *prowapi.ProwJob) runtime.Object { return in },
+			},
+			expected: sets.NewString("first"),
+		},
+		{
+			name: "no hidden repos returns all prowjobs except those not matching label selector",
+			prowJobs: []func(*prowapi.ProwJob) runtime.Object{
+				func(in *prowapi.ProwJob) runtime.Object {
+					in.Labels = map[string]string{}
+					return in
+				},
+			},
+			expected: sets.NewString(),
+		},
+		{
+			name: "hidden repos excludes prowjobs from those repos",
+			prowJobs: []func(*prowapi.ProwJob) runtime.Object{
+				func(in *prowapi.ProwJob) runtime.Object { return in },
+			},
+			hiddenRepos: sets.NewString("org/repo"),
+			expected:    sets.NewString(),
+		},
+		{
+			name: "hidden orgs excludes prowjobs from those orgs",
+			prowJobs: []func(*prowapi.ProwJob) runtime.Object{
+				func(in *prowapi.ProwJob) runtime.Object { return in },
+			},
+			hiddenRepos: sets.NewString("org"),
+			expected:    sets.NewString(),
+		},
+		{
+			name: "hidden repos excludes prowjobs from those repos even by extra_refs",
+			prowJobs: []func(*prowapi.ProwJob) runtime.Object{
+				func(in *prowapi.ProwJob) runtime.Object {
+					refs := *in.Spec.Refs
+					in.Spec.Refs = nil
+					in.Spec.ExtraRefs = []prowapi.Refs{refs}
+					return in
+				},
+			},
+			hiddenRepos: sets.NewString("org/repo"),
+			expected:    sets.NewString(),
+		},
+		{
+			name: "hidden orgs excludes prowjobs from those orgs even by extra_refs",
+			prowJobs: []func(*prowapi.ProwJob) runtime.Object{
+				func(in *prowapi.ProwJob) runtime.Object {
+					refs := *in.Spec.Refs
+					in.Spec.Refs = nil
+					in.Spec.ExtraRefs = []prowapi.Refs{refs}
+					return in
+				},
+			},
+			hiddenRepos: sets.NewString("org"),
+			expected:    sets.NewString(),
+		},
+		{
+			name: "hidden repos cant exclude prowjobs without refs",
+			prowJobs: []func(*prowapi.ProwJob) runtime.Object{
+				func(in *prowapi.ProwJob) runtime.Object {
+					in.Spec.Refs = nil
+					return in
+				},
+			},
+			hiddenRepos: sets.NewString("org/repo"),
+			expected:    sets.NewString("first"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		var data []runtime.Object
+		for _, generator := range testCase.prowJobs {
```

Comment:
this is pretty nifty, although why not just do `data = []runtime.Object{ testcase.prowJob(templateJob.DeepCopy()) }`?

Reply:
I think I wanted to have a slice of jobs in a test case but forgot to write that one :facepalm: